jep-014-string-functions.md

```diff
@@ -14,6 +14,12 @@ This JEP introduces a core set of useful string manipulation functions. Those fu
 
 ## Specification
 
+Some string manipulation functions bring the new concept of _optional arguments_ to JMESPath functions. The specification paragraph on function evaluation must thus be changed accordingly – highlighted in **bold** in the text below:
+
+_Functions can ~~either~~ have a specific arity, **a range of valid – minimum and maximum – number of arguments** or be variadic with a minimum number of arguments. If a function-expression is encountered where the arity does not match or the minimum number of arguments for a variadic function is not provided, then implementations must indicate to the caller tha~~n~~**t** an invalid-arity error occurred. How and when this error is raised is implementation specific. How and when this error is raised is implementation specific._
```

Comment:
Is the strikethrough + correction intentional? And if so, I wonder about the value of indicating past presence of a typographical error vs. e.g. documenting revision history.
```suggestion
_Functions can ~~either~~ have a specific arity, **a range of valid – minimum and maximum – number of arguments** or be variadic with a minimum number of arguments. If a function-expression is encountered where the arity does not match or the minimum number of arguments for a variadic function is not provided, then implementations must indicate to the caller that an invalid-arity error occurred. How and when this error is raised is implementation specific. How and when this error is raised is implementation specific._
```

Reply:
Yes it is intentional, although I certainly do not want to come across as petty pointing out minor typos.

Since the text has been frozen in time for so long, I figured typos could be fixed at the same time as doing the change.

In fact, I have been bitten perpetuating typos when incorporating the changes from some of those JEPs so that helps me above all.

Reply:
But yeah I kinda see your point.